### DISCUSSION TITLE
How to break a line between \begin and \end?

### DISCUSSION THREAD

**Discussion by kobayashi-dot** [KaTeX/KaTeX#3057] [Q&A] [answered]
I want to display a matrix [[a, b], [c, d]] but the following code doesn't work:

    <!DOCTYPE html>
    <head>
        <meta charset="utf-8">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.13.11/dist/katex.min.css" integrity="sha384-Um5gpz1odJg5Z4HAmzPtgZKdTBHZdw8S29IecapCSB31ligYPhHQZMIlWLYQGVoc" crossorigin="anonymous">
    
        <script src="https://cdn.jsdelivr.net/npm/katex@0.13.11/dist/katex.min.js" integrity="sha384-YNHdsYkH6gMx9y3mRkmcJ2mFUjTd0qNQQvY9VYZgQd7DcN7env35GzlmFaZ23JGp" crossorigin="anonymous"></script>
    </head>
    <body>
        <div id="math">
            $$\begin{matrix}
            a & b\\
            c & d
            \end{matrix}$$
        </div>
        <script>
            const katex = window.katex
            const math = document.getElementById('math')
            math.innerHTML = math.textContent
          .replace(/\$\$(.*?)\$\$/g, function (outer, inner) {
            return katex.renderToString(inner, { displayMode: true })
          })
          .replace(/\$(.*?)\$/g, function (outer, inner) {
            return katex.renderToString(inner, { displayMode: false })
          })
        </script>
    </body>

However, this works:

    <div id="math">
      $$\begin{matrix}a & b\\ c & d\end{matrix}$$
    </div>

In other words, I can't break the line between \begin and \end.

**Comment by edemaine** [comment, marked as answer]
This is because the regular expression character `.` doesn't match newline (it matches everything except newline).  You could use `[^]` to match everything.

Is there a particular reason you're not using the auto-render extension, which does the `$` matching for you?
